src/hooks/useENSRegistrationStepHandler.tsx

```diff
@@ -72,16 +80,17 @@ export default function useENSRegistrationStepHandler(observer = true) {
   ]);
 
   const watchCommitTransaction = useCallback(async () => {
+    if (observer) return;
+    const provider = await getProviderForNetwork();
     let confirmed = false;
-    const tx = await web3Provider.getTransaction(
-      registrationParameters?.commitTransactionHash || ''
-    );
-    const block = await web3Provider.getBlock(tx.blockHash || '');
+    const tx = await provider.getTransaction(commitTransactionHash || '');
+    if (!tx?.blockHash) return confirmed;
```

Comment:
if no blockhash we know the tx is not confirmed so we shouldn't try to ask for the block